### PR TITLE
Use new `@solid-primitives/i18n` package

### PR DESCRIPTION
**Pull request by thetarnav**
The i18n package is not published yet: https://github.com/solidjs-community/solid-primitives/pull/510

### CODE REVIEW

File: src/components/Nav/index.tsx [hunk 1]
```diff
@@ -257,20 +258,21 @@ const Nav: ParentComponent<{ showLogo?: boolean; filled?: boolean }> = (props) =
...
-                    'bg-solid-medium text-white': lang == locale(),
-                    'hover:bg-solid-light': lang == locale(),
-                  }}
-                  onClick={() => locale(lang) && toggleLangs(false)}
-                >
-                  {label}
-                </button>
-              )}
-            </For>
+            {entries(langs).map(([lang, label]) => (
```
Comment (davedbase) [inline src/components/Nav/index.tsx:261]: Wondering why you opted for .map here instead of <For>?

File: src/pages/404.tsx [hunk 2]
```diff
@@ -14,14 +14,12 @@ const FourOhFour: Component = () => {
...
-          <h2 class="mt-5 text-4xl font-semibold">
-            {t('global.404.header', {}, 'Oops. Four oh four.')}
-          </h2>
+          <h2 class="mt-5 text-4xl font-semibold">{t('global.404.header')}</h2>
           <h2 class="text-2xl text-solid-gray">
             <a class="text-solid-medium" href="https://github.com/solidjs/solid-router">
               @solidjs/router
             </a>{' '}
-            {t('global.404.body', {}, "believes this page definitely doesn't exist.")}
+            {t('global.404.body')}
```
Comment (davedbase) [inline src/pages/404.tsx:22]: Would it make sense to use the scoped primitive for situations like this?

**Comment by davedbase** [comment]
@modderme123 will these changes mesh with the changes you were planning for start? If so, how would you prefer we handle this?

File: src/components/Nav/index.tsx (hunk 1 above)
Comment (thetarnav) [reply src/components/Nav/index.tsx:261]: `<For each={Object.entries(obj)} />` will have the same effect as `Object.entries(obj).map()` so no point to do the diffing

File: src/pages/404.tsx (hunk 2 above)
Comment (thetarnav) [reply src/pages/404.tsx:22]: it's just syntax differece
this way is simpler imo

**Pull request merged**